### PR TITLE
Issue16

### PR DESCRIPTION
**Pull request by Taimoorrana1**
Removed TDG calls from update.py

### CODE REVIEW

File: app/core/update.py [hunk 1]
```diff
@@ -1,36 +1,41 @@
 from app.TDG import WaitingTDG
```
Comment (arekmano) [inline app/core/update.py]: WaitingTDG is still imported

File: app/core/update.py [hunk 2]
```diff
@@ -1,36 +1,41 @@
...
-						ReservationTDG.insertDirect(w[1], w[3], w[2], w[4])
-						WaitingTDG.delete(w[0])
-						return
-			i = i + 1
\ No newline at end of file
+# waitingList: list containing (waitingId, room, reservee, description, timeslotid, starttime, endtime) entries
+# availabilityList: availabilities for that room (array of strings)
+def updateWaiting(roomId, date, availabilityList):
+    i = 0
+    print(availabilityList)
```
Comment (arekmano) [inline app/core/update.py]: better to print log statements than print objects

File: app/core/update.py [hunk 3]
```diff
@@ -1,36 +1,41 @@
...
+    waitingList = WaitingMapper.findRoomOnDate(roomId, date)
+    if waitingList is None:
+        return
+    else:
+        for w in waitingList:
+            for a in availabilityList:
+                if a == "Available" and availabilityList[i + 1] == "unavailable" and (i + 10) != 21:
+                    print("first if")
```
Comment (arekmano) [inline app/core/update.py]: Confusing log

File: app/core/update.py [hunk 4]
```diff
@@ -1,36 +1,41 @@
...
+                    if w[5] == (i + 9) and w[6] == (i + 9):
+                        r = Reservation(w[1], w[2], w[4], w[3])
+                        ReservationMapper.save(r)
+                        WaitingMapper.delete(w)
+                        return
+                elif a == "Available" and availabilityList[i + 1] == "Available" and (i + 10) != 20:
+                    print("first elif")
+                    if w[5] == (i + 9) and w[6] == (i + 10):
+                        print("first elif if")
```
Comment (arekmano) [inline app/core/update.py]: same

File: app/core/update.py [hunk 5]
```diff
@@ -1,36 +1,41 @@
...
+            for a in availabilityList:
+                if a == "Available" and availabilityList[i + 1] == "unavailable" and (i + 10) != 21:
+                    print("first if")
+                    if w[5] == (i + 9) and w[6] == (i + 9):
+                        r = Reservation(w[1], w[2], w[4], w[3])
+                        ReservationMapper.save(r)
+                        WaitingMapper.delete(w)
+                        return
+                elif a == "Available" and availabilityList[i + 1] == "Available" and (i + 10) != 20:
+                    print("first elif")
```
Comment (arekmano) [inline app/core/update.py]: same

File: app/core/update.py [hunk 6]
```diff
@@ -1,36 +1,41 @@
...
+                    if w[5] == (i + 9) and w[6] == (i + 10):
+                        print("first elif if")
+                        r = Reservation(w[1], w[2], w[4], w[3])
+                        ReservationMapper.save(r)
+                        WaitingMapper.delete(w)
+                        return
+                elif (i + 10) == 21 and a == "Available":
+                    print("second elif")
```
Comment (arekmano) [inline app/core/update.py]: same

File: app/core/update.py [hunk 7]
```diff
@@ -1,36 +1,41 @@
...
+# waitingList: list containing (waitingId, room, reservee, description, timeslotid, starttime, endtime) entries
+# availabilityList: availabilities for that room (array of strings)
+def updateWaiting(roomId, date, availabilityList):
+    i = 0
+    print(availabilityList)
+    waitingList = WaitingMapper.findRoomOnDate(roomId, date)
+    if waitingList is None:
+        return
+    else:
+        for w in waitingList:
```
Comment (arekmano) [inline app/core/update.py]: please make variables descriptive rather than single characters

File: app/core/update.py [hunk 8]
```diff
@@ -1,36 +1,41 @@
...
+                    if w[5] == (i + 9) and w[6] == (i + 10):
+                        print("first elif if")
+                        r = Reservation(w[1], w[2], w[4], w[3])
+                        ReservationMapper.save(r)
+                        WaitingMapper.delete(w)
+                        return
+                elif (i + 10) == 21 and a == "Available":
+                    print("second elif")
+                    if w[5] == 21 and w[6] == 21:
+                        print("second elif if")
```
Comment (arekmano) [inline app/core/update.py]: same

Comment (Taimoorrana1) [inline app/core/update.py]: All these code was there before. But you are right, I should have cleaned it.

**Pull request merged**